packages/functional-tests/tests/syncV3/signIn.spec.ts

```diff
@@ -20,129 +21,176 @@ test.describe('severity-2 #smoke', () => {
-      const { configPage, page, login, connectAnotherDevice, signinTokenCode } =
+    test('verified email, can ask to resend code', async ({
+      emails,
+      target,
+      syncBrowserPages,
+    }) => {
+      const { page, login, connectAnotherDevice, signinTokenCode } =
         syncBrowserPages;
+
       const [email] = emails;
-      const config = await configPage.getConfig();
-      test.fixme(
-        config.showReactApp.signUpRoutes,
-        'this test goes through the signup flow instead of sign in, skipping for react, see FXA-8973'
-      );
+
+      // Simulate a new sign up that requires a signin verification code.
+      target.auth.signUp(email, PASSWORD, {
+        service: 'sync',
+        preVerified: 'true',
+      });
 
       await page.goto(
         `${target.contentServerUrl}?context=fx_desktop_v3&service=sync&action=email`
       );
-      await login.setEmail(email);
-      await signinTokenCode.clickSubmitButton();
-      await login.setPassword(PASSWORD);
-      await login.confirmPassword(PASSWORD);
-      await login.setAge('21');
-      await login.submit();
+      await login.login(email, PASSWORD);
 
       // Click resend link
       await signinTokenCode.resendLink.click();
-      await signinTokenCode.successMessage.waitFor({ state: 'visible' });
       await expect(signinTokenCode.successMessage).toBeVisible();
       await expect(signinTokenCode.successMessage).toContainText(
         /Email re-?sent/
       );
       const code = await target.email.waitForEmail(
         email,
-        EmailType.verifyShortCode,
-        EmailHeader.shortCode
+        EmailType.verifyLoginCode,
+        EmailHeader.signinCode
       );
-      await signinTokenCode.input.fill(code);
-      await login.submit();
+      await signinTokenCode.submitCode(code);
       await expect(connectAnotherDevice.fxaConnected).toBeVisible();
     });
 
-    test('verified - invalid code', async ({
+    test('verified email with signin verification, accepts valid sign in code', async ({
       emails,
       target,
       syncBrowserPages,
     }) => {
-      const { configPage, page, login, connectAnotherDevice, signinTokenCode } =
+      const { page, login, connectAnotherDevice, signinTokenCode } =
         syncBrowserPages;
+
       const [email] = emails;
-      const config = await configPage.getConfig();
-      test.fixme(
-        config.showReactApp.signUpRoutes,
-        'this test goes through the signup flow instead of sign in, skipping for react, see FXA-8973'
+
+      // Simulate a new sign up that requires a signin verification code.
+      target.auth.signUp(email, PASSWORD, {
+        service: 'sync',
+        preVerified: 'true',
+      });
+
+      await page.goto(
+        `${target.contentServerUrl}?context=fx_desktop_v3&service=sync&action=email`
+      );
+      await login.login(email, PASSWORD);
+
+      const code = await target.email.waitForEmail(
```

Comment:
I've pulled these wait for email operations out of the POM. Previously, in some tests they would have this action embedded in a login page call, but sometimes would not have it embedded. 

Since this was already somewhat inconsistent, I decide to pull this out of the POM.  In my opinion, the test is more understandable if this step is clearly declared. I also think that the POM's primary responsibility should be interacting with a web page, and this is not directly a POM type concern/responsibility.